lib.typ

```diff
@@ -94,22 +106,34 @@
   ]
 }
 
-#let __resume_footer(author, language, lang_data, date) = {
+#let __resume_footer(author, language, lang_data, date, disable-smallcaps) = {
   set text(
     fill: gray,
     size: 8pt,
   )
   __justify_align_3[
-    #smallcaps[#date]
+    #if not disable-smallcaps [#smallcaps(date)] else [#date]
   ][
-    #smallcaps[
-      #if language == "zh" or language == "ja" [
-        #author.firstname#author.lastname
-      ] else [
-        #author.firstname#sym.space#author.lastname
-      ]
-      #sym.dot.c
-      #linguify("resume", from: lang_data)
+    #if not disable-smallcaps [
+      #smallcaps(
+        {
+          let name = if language == "zh" or language == "ja" {
+            str(author.firstname) + str(author.lastname)
+          } else {
+            str(author.firstname) + " " + str(author.lastname)
+          }
+          name + " · " + linguify("resume", from: lang_data)
+        }
+      )
+    ] else [
+      {
+        let name = if language == "zh" or language == "ja" {
+          str(author.firstname) + str(author.lastname)
+        } else {
+          str(author.firstname) + " " + str(author.lastname)
+        }
+        name + " · " + linguify("resume", from: lang_data)
```

Comment:
Could you factor the common code out here into a separate function and then conditionally wrap that in small caps or not depending on the flag? Not a fan of the duplicate code here.